File: apps/frontend/src/features/document/types/document.ts
export type LinkTargetType = "project" | "task";

export interface DocumentLink {
  id: string;
  target_type: LinkTargetType;
  target_id: string;
  created_at: string;
}

export interface DocumentAttachment {
  id: string;
  filename: string;
  content_type: string;
  size_bytes: number;
  created_at: string;
}

export interface Document {
  id: string;
  org_id: string;
  title: string;
  description?: string;
  created_by?: string;
  created_at: string;
  updated_at: string;
  links: DocumentLink[];
  attachments: DocumentAttachment[];
}

File: apps/backend/src/modules/knowledge/domain/entities/document.py
from dataclasses import dataclass, field
from uuid import UUID

from src.modules.knowledge.domain.entities.attachment import Attachment
from src.modules.knowledge.domain.entities.document_link import DocumentLink
from src.modules.shared.domain.entity_v1 import Entity, TimestampedMixin


@dataclass(eq=False)
class Document(Entity, TimestampedMixin):
    org_id: UUID
    title: str
    description: str | None = None
    created_by: UUID | None = None
    links: list[DocumentLink] = field(default_factory=list)
    attachments: list[Attachment] = field(default_factory=list)

    def update_content(self, title: str | None, description: str | None) -> None:
        if title is not None:
            self.title = title
        if description is not None:
            self.description = description
        self.touch()
